docs/en/development/io_processing.md

```diff
@@ -3,7 +3,51 @@ sidebar_label: Read/Write/Delete Processing Flow
+
+With the writeback mode enabled, JuiceFS will aggressively preserve all of the data into the cache directory without checking the size of the objects to be uploaded by default. This is useful in some scenarios which generate a huge plenty of intermediate files such as compiling software. In addition, a new parameter `--upload-delay` has been introduced since JuiceFS v0.17, which is for delaying the upload action to locally cache the data in a more aggressive way. If a file has been deleted before uploading, the upload action will ignore this file. This is not only a performance improvement, but also a cost saving. Meanwhile, comparing with local disk, JuiceFS uploads files automatically when the cache directory is running out of space, which keeps the applications away from unexcepted failures. This is useful for the applications requiring temporary file storages such as Spark shuffle.
+
+## Workflow of processing read
+
+JuiceFS reads objects from object storage block by block in a size of 4 MiB, which could be considered as a kind of read-ahead. Data read will be saved into the local cache directory for future use, proving by the high write bandwidth of blockcache in the 2nd highlighted section in the metrics figure. Obviously, the cached data will all be accessed by consequent reads. The high cache hit rate results in the maximization of the object storage read/write performance. In this case, the dataflow along each component shows as follows:
+
+![read](../images/internals-read.png)
+
+> **NOTICE**: When a read object arrives, if the decryption and compression are enabled, it will be first decrypted and then uncompressed by JuiceFS client, which is contrary to write.
+
+This strategy is not effective for randomly reading a small block within a large file. The read magnification will lower the system utilization, so as to the frequent write and evict to the local cache. Unfortunately, it is hard to benefit a lot from a cache system in this case. One possible solution is to enlarge the total capacity of the cache system to cache as much data as possible, while another solution could be disabling cache by setting `--cache-size 0` and using a object storage whose performance is as high as possible.
+
+Reading small files is much more easier. We often read the whole file in a single request. Since small files are cached first before writing to the object storage, if a file is read soon after it is written, the read operation will hit the local cache, like what JuiceFS bench does. The performance will be impressive.
```

Comment:
```suggestion
Reading small files is much easier. The whole file is often read in a single request. Since small files are cached first before writing to an object storage, if a file is read soon after it is written, the read operation will hit the local cache, like what JuiceFS bench does, and an impressive performance can be expected.
```